GVFS/GVFS.Platform.Windows/WindowsPlatform.cs

```diff
@@ -220,25 +220,28 @@ public override void ConfigureVisualStudio(string gitBinPath, ITracer tracer)
             try
             {
                 const string GitBinPathEnd = "\\cmd\\git.exe";
-                string[] gitVSRegistryKeyNames =
-                {
-                    "HKEY_CURRENT_USER\\Software\\Microsoft\\VSCommon\\15.0\\TeamFoundation\\GitSourceControl",
-                    "HKEY_CURRENT_USER\\Software\\Microsoft\\VSCommon\\16.0\\TeamFoundation\\GitSourceControl"
-                };
+                const string VSRegistryKeyRoot = @"Software\Microsoft\VSCommon";
+                const string GitVSRegistrySubKey = @"TeamFoundation\GitSourceControl";
                 const string GitVSRegistryValueName = "GitPath";
 
                 if (!gitBinPath.EndsWith(GitBinPathEnd))
                 {
-                    tracer.RelatedWarning(
-                        "Unable to configure Visual Studio’s GitSourceControl regkey because invalid git.exe path found: " + gitBinPath,
-                        Keywords.Telemetry);
-
                     return;
                 }
 
                 string regKeyValue = gitBinPath.Substring(0, gitBinPath.Length - GitBinPathEnd.Length);
-                foreach (string registryKeyName in gitVSRegistryKeyNames)
+
+                /* Get all versions of Visual Studio that exist in the registry at least 15.0.
+                 * This attempts to future proof (current version is 17.0), but may need to be
+                 * revisited in the future if VS changes how it stores this. */
+                var vsVersions = Registry.CurrentUser.OpenSubKey(VSRegistryKeyRoot)
+                    ?.GetSubKeyNames()
+                    .Where(name => float.TryParse(name, out var version) && version >= 15.0)

```

Comment:
The equality arm of `>= 15.0` unnerves me.. comparing equality of two floating-point numbers. But, if this has been tested for VS15, then I guess this is ok.

Reply:
Could also use `System.Version.TryParse` and compare the `.Major` version parts.